mantidproject/mantid
# Reorganize tests of fit functions, category general, chunk 2

This is now ready for testing:
- [x] Merge #16711 first.

Moves the unit test cases that run the Fit algorithm to FitTest, and reorganizes the function tests.

**To test:**
- Code review
- Tests should pass on all platforms. Some tests that were disabled have been re-enabled.
- Ensure that:
  - The algorithm `Fit` is not used in the function unit tests, and the old cases that were using `Fit` are now in `FitTest`
  - No additional dependencies/headers/using are added in `FitTest`, although there is an exception with `PawleyFunction` that I think is not easy to avoid.
  - Other changes remove unneeded headers, prevent leaks, use initialization lists, etc.

Fixes #16484.
This is the second last step of #16267

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
